wandb/sdk/wandb_settings.py

```diff
@@ -1864,6 +1871,16 @@ def _apply_init(self, init_settings: Dict[str, Union[str, int, None]]) -> None:
                         "Tried to auto resume run with "
                         f"id {resume_run_id} but id {self.run_id} is set.",
                     )
+
+        # handle rewind logic
+        if self.resume_from is not None:
+            if self.run_id is not None:
+                wandb.termwarn(
+                    "You cannot specify both run_id and resume_from. "
+                    "Ignoring run_id."
+                )
+            self.update({"run_id": self.resume_from.run}, source=Source.INIT)
+
```

Comment:
is it ok that the run_id could be overwritten later on in the function?

Reply:
i dont think it could right? because on the only other subsequent `self.update` line that could overwrite, the argument is `self.run_id or generate_id()`, but since we set `self.run_id` on this line, that subsequent line will always be a no op